src/envoy/utils/utils.cc

```diff
@@ -70,6 +76,21 @@ bool GetIpPort(const Network::Address::Ip* ip, std::string* str_ip, int* port) {
   return false;
 }
 
+bool GetDestinationUID(const envoy::api::v2::core::Metadata& metadata,
+                       std::string* uid) {
+  const auto filter_it = metadata.filter_metadata().find(kPerHostMixer);
+  if (filter_it == metadata.filter_metadata().end()) {
+    return false;
+  }
+  const auto fields_it =
+      filter_it->second.fields().find(kMetadataDestinationUID);
+  if (fields_it == filter_it->second.fields().end()) {
+    return false;
+  }
+  *uid = fields_it->second.string_value();
```

Comment:
may need to check second.type to make sure it is string case

Reply:
I think this is fine even if it is not a string. It is a user error to put other values, and this defaults to an empty string.